File: algorithms/binary_search.py
def binary_search(sorted_list, value):
    """Given a sorted list and a value, returns the value's index.

    Positional arguments:
    sorted_list -- a list sorted from low to high, e.g. [1, 3, 5, 9].
    """
    low = 0
    high = len(sorted_list) - 1
    while low <= high:
        middle = (high + low) // 2
        if value < sorted_list[middle]:
            high = middle - 1
        elif value > sorted_list[middle]:
            low = middle + 1
        else:
            return middle
    return None

File: src/algorithms/binary_search.c
/**
* binary_search - continually halve an array to find the index of an element
*
* @numbers: the array to search through
* @count: the length of the array
* @number: the number to find in the array
*
* Returns: the index of the number if found, else -1
*/
int binary_search(int numbers[], unsigned int count, int number) {
    int low = 0;
    int high = count - 1;

    while (low <= high) { 
        int middle = (low + high) / 2;
        if (number > numbers[middle]) {
            low = middle + 1;
        } else if (number < numbers[middle]) {
            high = middle - 1;
        } else {
            return middle;
        }
    }
    return -1;
}
